### PR TITLE
Create login page

### PR DESCRIPTION
**Pull request by Gerasko-Vadim**

### CODE REVIEW

File: src/modals/LoginModal.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                <div className="relative mt-6 flex-1 px-4 sm:px-6">
+                  <Input
+                    label='Enter your email address'
+                    placeholder='Enter your email address'
+                  />
+                  <Input
+                    label='Password'
+                    placeholder='Password'
+                  />
+                  <Button className='w-full mt-[24px]'>
```
Comment (Emmerhaus) [inline src/modals/LoginModal.tsx:25]: nit: Why in some cases there are `mt-6` and in some `mt-[24px]`? I would suggest keeping consistency and using px only for extreme values.

File: src/modals/LoginModal.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                    Login
+                  </Button>
+                  <Button className='w-full mt-[24px]'>
+                    Forgotten your password?
```
Comment (Emmerhaus) [inline src/modals/LoginModal.tsx:29]: nit: I usually see in this form "Forgot", 'Forgotten' sounds odd

File: src/components/ui/Input.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+export default function Input({
+  label,
+  classNameInput,
+  classNameLabel,
+  ...rest }: InputProps) {
+  const id = useId()
+
+  const classNameInputMerge = "h-[54px] bg-secondary border border-gray-300 text-gray-900 text-L rounded-lg focus:ring-brand-solid-hover block w-full p-2.5 "
```
Comment (Emmerhaus) [inline src/components/ui/Input.tsx:16]: I would suggest to use clsx instead or tailwind merge

**Review by Emmerhaus — COMMENTED** [review]
few nitpicks, lgtm in general

File: src/components/ui/Input.tsx (hunk 3 above)
Comment (iakivpekarskyi) [inline src/components/ui/Input.tsx:16]: Your are using "text-sm"  , "text-L" It's a good idea to use consistent naming conventions for better readability and maintainability.

**Review by iakivpekarskyi — COMMENTED** [review]
Reviewed

File: src/modals/LoginModal.tsx (hunk 2 above)
Comment (Gerasko-Vadim) [reply src/modals/LoginModal.tsx:29]: Yes, but that's how it's designed

File: src/app/page.tsx [hunk 4]
```diff
@@ -6,6 +7,7 @@ export default function Home() {
     <>
       <Hero />
       <ProductList />
+      <LoginModal/>
     </>
   )
 }
```
Comment (Wdyffs) [inline src/app/page.tsx:13]: Login modal should be connected with cart button in the header with open close functionality

File: src/components/ui/Input.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+          {label}
+        </label>
+      }
+      <input
+        type="text"
+        id={id}
+        className={classNameInputMerge}
+        {...rest}
+      />
+    </div>
```
Comment (Wdyffs) [inline src/components/ui/Input.tsx:37]: Input component will be used in different page with react-hook-form.
You should add error element (like div) that will accept error message, that will be provided into every input component

File: src/components/ui/Input.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import { HTMLAttributes, useId } from 'react'
+
+type InputProps = HTMLAttributes<HTMLInputElement> & {
+  label?: string,
+  classNameInput?: string,
+  classNameLabel?: string
+}
```
Comment (Wdyffs) [inline src/components/ui/Input.tsx:7]: Should improve types for the next integration with react-hook-form

File: src/components/ui/Input.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+export default function Input({
+  label,
+  classNameInput,
+  classNameLabel,
+  ...rest }: InputProps) {
+  const id = useId()
```
Comment (Wdyffs) [inline src/components/ui/Input.tsx:14]: Component should accept 'id' prop rather then use 'useId' hook

File: src/modals/LoginModal.tsx [hunk 8]
```diff
@@ -0,0 +1,47 @@
+import Button from '@/components/ui/Button';
+import Input from '@/components/ui/Input';
+
+export default function LoginModal() {
+  return (
+    <div className="relative z-10" aria-labelledby="slide-over-title" role="dialog" aria-modal="true">
+      <div className="fixed inset-0 bg-gray-500 bg-opacity-75 transition-opacity"></div>
+      <div className="fixed inset-0 overflow-hidden">
```
Comment (Wdyffs) [inline src/modals/LoginModal.tsx:8]: You should use 'form' as parent element because of use react-hook-form

**Review by Wdyffs — CHANGES_REQUESTED** [review]
Create form with react-hook-form
Custom input types should be adopt to accept react-hook-form props
Add validation to email and password

**Pull request closed without merging**